eo-runtime/src/main/java/org/eolang/ExprReduce.java

```diff
@@ -74,71 +67,138 @@
      * Ctor.
      *
      * @param oper Operation name
-     * @param param Name of parameter with varargs
-     * @param type Type of varargs
-     * @param reduction Reduction operation on consecutive varags
-     * @param validation Validation function
+     * @param param Param name with varargs
+     * @param arguments Object contains type of varargs, validation and reduction functions
      * @checkstyle ParameterNumberCheck (10 lines)
      */
     public ExprReduce(
         final String oper,
         final String param,
-        final Class<T> type,
-        final BiFunction<T, T, T> reduction,
-        final Function<T, String> validation
+        final Args<T> arguments
     ) {
-        this.param = param;
-        this.type = type;
-        this.reduction = reduction;
         this.oper = oper;
-        this.validation = validation;
-    }
-
-    /**
-     * Ctor.
-     * @param oper Peration name
-     * @param param Name of parameter with varargs
-     * @param type Type of varargs
-     * @param reduction Reduction operation on consecutive varargs
-     * @checkstyle ParameterNumberCheck (10 lines)
-     */
-    public ExprReduce(
-        final String oper,
-        final String param,
-        final Class<T> type,
-        final BiFunction<T, T, T> reduction
-    ) {
-        this(
-            oper,
-            param,
-            type,
-            reduction,
-            x -> ""
-        );
+        this.param = param;
+        this.arguments = arguments;
     }
-
     @Override
     public Phi get(final Phi rho) {
-        T acc = new Param(rho).strong(this.type);
+        T acc = this.arguments.accumulator(rho);
         final Phi[] args = new Param(rho, this.param).strong(Phi[].class);
         for (int idx = 0; idx < args.length; ++idx) {
```

Comment:
@potatmen This whole loop can be part of `Args.get()` method.

Reply:
@mximp but in this case we move almost the whole functionality of  `get()` method to methods of `Arg<> object`, is it ok?

Reply:
@potatmen we move only logic related extracting and validating arguments which is fine. `Args<>` object remains concise. `ExprReduce.get()` still does reduction on prepared arguments.

Reply:
@mximp Ok, for now I see this structure of `ExprReduce.get()`:
```
T acc = new Param(rho).strong(this.type);
List<T> validated = Args.get(rho, this.param (it's used to make Phi [] args array));
for (val in validated) {
   acc = this.reduction(acc,val);
}
return ...
```
In this structure `ExprReduce `has 3 attributes: `param, Args<T>, reduction function`
And Args has 3 attributes: `oper, varargs type, validation function` 
Also, `Args `has one method `get(Phi rho, String param)` that extracts and validates args.

Reply:
Still thinking about how to make Args to have only one method

Reply:
I can do it in this way, `Args.get()` will look like this:
```
T acc = new Param(rho).strong(this.type);
final Phi[] args = new Param(rho, param).strong(Phi[].class);
List<T> ls = new ArrayList<T>();
ls.add(acc);
*Arguments extraction*
return ls;
```
And the new look of `ExprReduce.get()`:
```
List<T> validated = Args.get(rho, this.param (it's used to make Phi [] args array));
T acc = validated.get(0);
for (val in validated) {
   acc = this.reduction(acc,val);
}
return ...
```

Reply:
In this case Args is going to have one method

Reply:
@mximp

Reply:
@potatmen You can make `Args.get()` accept only single parameter. Others can be supplied in constructor.
Also in your suggested code you will reduce `acc` twice. You can use `Stream.reduce()` as I mentioned above which handles this situation.

Reply:
@mximp got your idea
Btw, shlouldn't it look like this `acc = this.args.get(rho).stream().reduce(acc, this.reduction);` ?

Reply:
@potatmen In this case you need to have `acc` in advance which complicates things.

Reply:
@mximp I believe `reduce()` should accept 2 arguments
```
T reduce(T identity, BinaryOperator<T> accumulator);

Where, identity is initial value 
of type T and accumulator is a 
function for combining two values.
```

Reply:
@potatmen 
https://docs.oracle.com/javase/8/docs/api/java/util/stream/Stream.html#reduce-java.util.function.BinaryOperator-